### PR TITLE
Add decodeFirstPage

### PR DESCRIPTION
**Pull request by sz-piotr**

### CODE REVIEW

File: packages/encoding/src/readFundingIndices.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { ByteReader } from './ByteReader'
+import { MIN_INT } from './constants'
+import { decodeAssetId } from './decodeAssetId'
+import { FundingIndex } from './OnChainData'
+
+export function readFundingIndices(reader: ByteReader) {
+  const fundingIndicesLength = reader.readNumber(32)
+  const indices: FundingIndex[] = []
+  for (let i = 0; i < fundingIndicesLength; i++) {
+    reader.skip(17)
```
Comment (hasparus) [inline packages/encoding/src/readFundingIndices.ts:10]: `reader.skip` mutates, so we need to call these functions in a set order, right?

File: packages/encoding/src/readState.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { ByteReader } from './ByteReader'
+import { decodeState } from './decodeState'
+
+export function readState(reader: ByteReader) {
```
Comment (hasparus) [inline packages/encoding/src/readState.ts:4]: Do we have some docs about this data layout that we could mention in the readme/comments?

File: packages/encoding/test/decodeAssetId.test.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { expect } from 'chai'
 
-import { decodeAssetId, encodeAssetId } from '../src/assetId'
+import { decodeAssetId, encodeAssetId } from '../src/decodeAssetId'
```
Comment (hasparus) [inline packages/encoding/test/decodeAssetId.test.ts:3]: nit: I'm not very attached to file names personally, but `encodeAssetId` is now the living in the shadow of its inverse function :P

File: packages/encoding/test/decodeUpdates.test.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+      DecodingError,
+      'Went out of bounds'
+    )
+  })
+
+  it('decodes multiple entries and positions', () => {
+    const writer = new ByteWriter()
+      .writeNumber(2, 32) // 2 entries
+      .writeNumber(3, 32) // 3 indices
+      .writePadding(17)
```
Comment (hasparus) [inline packages/encoding/test/decodeUpdates.test.ts:21]: Very neat chaining.

**Comment by hasparus** [comment]
Whenever a number is signed, we add MIN_INT, right? How do we know which ones are signed? Do we have some docs on this encoding to refer to?

**Review by hasparus — APPROVED** [review]

File: packages/encoding/src/readFundingIndices.ts (hunk 1 above)
Comment (sz-piotr) [reply packages/encoding/src/readFundingIndices.ts:10]: Yup

File: packages/encoding/test/decodeAssetId.test.ts (hunk 3 above)
Comment (sz-piotr) [reply packages/encoding/test/decodeAssetId.test.ts:3]: Oh you're right. I'm going to change the file name

**Comment by sz-piotr** [comment]
The encoding is defined in the python codebase we've been given

**Pull request merged**